File: lib/features/chat/models/conversation.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'message.dart';

part 'conversation.freezed.dart';
part 'conversation.g.dart';

@freezed
class Conversation with _$Conversation {
  const factory Conversation({
    required int id,
    required bool isGroup,
    String? title,
    required String createdBy,
    required DateTime createdAt,
    DateTime? lastMessageAt,
    @Default([]) List<Participant> participants,
    Message? lastMessage,
    @Default(0) int unreadCount,
  }) = _Conversation;

  factory Conversation.fromJson(Map<String, dynamic> json) =>
      _$ConversationFromJson(json);
}

@freezed
class Participant with _$Participant {
  const factory Participant({
    required String userId,
    String? role,
    required DateTime joinedAt,
    String? userName,
    String? userAvatar,
  }) = _Participant;

  factory Participant.fromJson(Map<String, dynamic> json) =>
      _$ParticipantFromJson(json);
}

File: MoviePlusBackend/MoviePlusApi/Models/Chat/Conversation.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MoviePlusApi.Models.Chat
{
    public class Conversation
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        public bool IsGroup { get; set; } = false;
        
        [MaxLength(200)]
        public string? Title { get; set; }
        
        [Required]
        public Guid CreatedBy { get; set; }
        
        [Required]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        public DateTime? LastMessageAt { get; set; }
        
        // Navigation properties
        public virtual ICollection<ConversationParticipant> Participants { get; set; } = new List<ConversationParticipant>();
        public virtual ICollection<Message> Messages { get; set; } = new List<Message>();
    }
}